### PR TITLE
fix admin login after search

### PR DESCRIPTION
**Pull request by Maroo-b**
WIP PR for fixing admin login after search more on (https://www.pivotaltracker.com/story/show/116052695)

- Submit regression test to replicate the issue

### CODE REVIEW

**Comment by tansaku** [comment]
@Maroo-b is this now good to merge?

**Comment by Maroo-b** [comment]
@tansaku as I checked locally it fix the described issue, did you check it from your side ?

**Comment by tansaku** [comment]
checking it locally now @Maroo-b

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -45,7 +45,7 @@ def after_sign_in_path_for(resource)
     set_flash_warning_reminder_to_update_details resource
     return edit_user_path id: current_user.id if session[:pending_organisation_id]
     return organisation_path(current_user.organisation) if current_user.organisation
-    return session[:previous_url] if session[:previous_url]
+    return session[:previous_url] if session[:previous_url] && session[:previous_url] != organisations_search_path
```
Comment (tansaku) [inline app/controllers/application_controller.rb]: so I might break this out into a separate method that explains what we are doing.  Exceeding the line length limit is often a hint that the line is going to be difficult to read, so how about:

```ruby
return session[:previous_url] if appropriate_previous_url_available?
```

```ruby
def appropriate_previous_url_available?
  session[:previous_url] && session[:previous_url] != organisations_search_path
end
```

Comment (Maroo-b) [inline app/controllers/application_controller.rb]: Thank you @tansaku  for pointing this out, I'll update the PR

**Comment by Maroo-b** [comment]
@tansaku  PR updated with the refactroing.

**Comment by tansaku** [comment]
great work on this @Maroo-b !

**Pull request merged**